internal/databroker/server.go

```diff
@@ -225,10 +226,19 @@ func (srv *Server) doSync(ctx context.Context, recordVersion *string, db storage
 		return updated[i].Version < updated[j].Version
 	})
 	*recordVersion = updated[len(updated)-1].Version
-	return stream.Send(&databroker.SyncResponse{
-		ServerVersion: srv.version,
-		Records:       updated,
-	})
+	for i := 0; i < len(updated); i += syncBatchSize {
```

Comment:
I'm not clever enough to know just by reading this if there could be an off by one error in this for loop. Do you mind adding some tests ? :)

Reply:
Added.